webui/react/src/components/Navigation.tsx

```diff
@@ -50,14 +50,16 @@ const Navigation: React.FC<Props> = ({ children }) => {
   const rbacEnabled = useFeature().isOn('rbac'),
     mockAllPermission = useFeature().isOn('mock_permissions_all'),
     mockReadPermission = useFeature().isOn('mock_permissions_read');
-  usePolling(
-    () => {
-      if (rbacEnabled && !mockAllPermission && !mockReadPermission && currentUser !== NotLoaded) {
-        fetchMyRoles();
-      }
-    },
-    { interval: 120000 },
-  );
+  const syncRoles = useCallback(() => {
+    if (rbacEnabled && !mockAllPermission && !mockReadPermission && currentUser !== NotLoaded) {
+      fetchMyRoles();
+    }
+  }, [currentUser, fetchMyRoles, rbacEnabled, mockAllPermission, mockReadPermission]);
+
+  useEffect(() => {
+    syncRoles();
+  }, [currentUser, syncRoles]);
```

Comment:
**nit chore:** this `currentUser` dependency is redundant because it is already a dependency for `syncRoles`.

Reply:
thanks!